nifi-extension-bundles/nifi-py4j-bundle/nifi-py4j-bridge/src/main/java/org/apache/nifi/py4j/PythonProcess.java

```diff
@@ -302,8 +302,14 @@ String resolvePythonCommand() throws IOException {
         } else if (virtualEnvDirectories.length == 1) {
             commandExecutableDirectory = virtualEnvDirectories[0].getName();
         } else {
-            // Default to bin directory for macOS and Linux
-            commandExecutableDirectory = "bin";
+            // We should get at most 2 directories.  Check for python command.
+            if (virtualEnvDirectories[0].list((file, name) -> name.startsWith(pythonCmd)).length >= 1) {
+                commandExecutableDirectory = virtualEnvDirectories[0].getName();
+            } else if (virtualEnvDirectories[1].list((file, name) -> name.startsWith(pythonCmd)).length >= 1) {
+                commandExecutableDirectory = virtualEnvDirectories[1].getName();
+            } else {
+                throw new IOException("Failed to find pythond command: " + pythonCmd);
```

Comment:
Minor spelling and string construction recommendation:
```suggestion
                throw new IOException("Failed to find Python command [%s]".formatted(pythonCmd));
```

Reply:
Thank you for catching that and for the additional feedback of course.